<launch>
	<arg name="remapping_disabled" default="false"/>
	<arg name="cameras_prefix" default="cameras"/>
	<arg name="camera_name" default="PiCam"/>
	<arg name="camera_ns" default="/$(arg cameras_prefix)/$(arg camera_name)"/>

	<node pkg="cv_camera" type="cv_camera_node" name="cv_camera_node">
		<param name="frame_id" value="$(arg camera_name)" />
		<param name="device_path" value="/dev/video0" />
		<param name="rate" value="10.0" />
		<param name="image_width" value="640" />
		<param name="image_height" value="480" />
		<param name="camera_info_url" value="file://$(find aruco_analyzer)/config/$(arg camera_name).yaml"/>
		<remap if="$(eval arg('remapping_disabled') == false)" from="/cv_camera_node/image_raw" to="$(arg camera_ns)/image_raw"/>
		<remap from="/$(arg camera_name)/camera_info" to="$(arg camera_ns)/camera_info"/>	
	</node>
</launch>
